libqtile/backend/base/window.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABCMeta
+
+
+class Window(metaclass=ABCMeta):
```

Comment:
If we're introducing these, it seems like we should make window.Window a subclass and change the tests for window.Window. Otherwise it's just dead code.

Reply:
For now, it would be. I'm finding it difficult to identify the best order of changes to make to move `window.Window` into the backend.

Moving method at a time from `window.Window` to `xcbw.Window` could work but would be ugly -- any usage of a given method would be converted from `window.method` to `window.window.method` until they are all moved and then they are merged properly.

Alternatively there is moving it all at once, which I submitted previously and was a big diff.

Either way, introduction of the base classes as per this addition isn't necessary until after the merge, so I'd be fine with not introducing it just yet.